### ISSUE TITLE
ptr is not defined for this document 

### ISSUE THREAD

**Issue by mohsen1** [apigee-127/swagger-tools#173]
https://github.com/whitlockjc/json-refs/issues/8

**Comment by nickretallack** [comment]
There are various bugs in the implementation of ```allOf``` when it is encountered in a child of the ```definitions``` block.  However, It works fine inside a ```schema``` block.

[According to the spec](https://github.com/swagger-api/swagger-spec/blob/master/versions/2.0.md#definitions-object), children of ```definitions``` ``are`` ```schema``` blocks.  Why are they treated differently here?

**Comment by whitlockjc** [comment]
I'm not sure.  I would had expected the code @mohsen1 posted to give a schema error because it didn't look like a valid schema object to me.  I use the Swagger JSON Schema to validate all of the document and only after that do I validate the semantics, none of which is related to model structure since that should be handled by the Swagger JSON Schema file.  I'm not sure swagger-tools could do more but if you disagree or have ideas, let me know.

**Comment by mohsen1** [comment]
the `.validate` call should not ever throw an error if I'm passing an object. It's acceptable if I pass `undefined` or `null` you throw error but when I pass object, it shouldn't throw. Because I don't want to wrap the `.validate` in `try/catch`.

It should call the callback with the error.

**Comment by whitlockjc** [comment]
You shouldn't have to worry about that anymore.  I will file a bug to fix this in the long term.

**Comment by mohsen1** [comment]
works for me. I'm closing the upstream bug since the schema is invalid.

**Comment by nickretallack** [comment]
How is the schema invalid?  What's the problem with it?

**Comment by whitlockjc** [comment]
The schema isn't invalid at all, I just don't know what you'd expect it to do.  Based on the YAML I was given, I did find a bug and I committed a fix for it above.  Here is the YAML:

```yaml
swagger: '2.0'
info:
  version: "0.0.0"
  title: Example
paths:
  /test:
    get:
      responses:
        200:
          description: OK
definitions:
  Two:
    allOf:
      - $ref: '#/definitions/One' 
      - format: something
  One:
    type: object
```

`#/definitions/Two/allOf/1` makes no sense from a definition/model composition perspective.  So while it's not invalid, it won't do much for you either.  Swagger uses the `properties` object for a `schema` object for definition/model composition from my understanding.

I've cc'd @webron to get his take as well.
